enhancements/workload-partitioning/wide-availability-workload-partitioning.md

```diff
@@ -0,0 +1,407 @@
+---
+title: wide-availability-workload-partitioning
+authors:
+  - "@eggfoobar"
+reviewers:
+  - "@jerpeter1"
+approvers:
+  - "@jerpeter1"
+api-approvers:
+  - None
+creation-date: 2022-08-03
+last-updated: 2022-08-09
+tracking-link:
+  - https://issues.redhat.com/browse/CNF-5562
+see-also:
+  - "/enhancements/workload-partitioning"
+  - "/enhancements/node-tuning/pao-in-nto.md"
+---
+
+# Wide Availability Workload Partitioning
+
+## Summary
+
+This enhancements builds on top of the [Management Workload
+Partitioning](management-workload-partitioning.md) and the [move of PAO in
+NTO](../node-tuning/pao-in-nto.md) enhancement to provide the ability to do
+workload partitioning to our wider cluster configurations. The previous workload
+partitioning work is isolated to Single Node cluster configurations, this
+enhancement will seek to allow customers to configure workload partitioning on
+HA as well as Compact(3NC) clusters.
+
+## Motivation
+
+Customers who want us to reduce the resource consumption of management workloads
+have a fixed budget of CPU cores in mind. We want to use normal scheduling
+capabilities of kubernetes to manage the number of pods that can be placed onto
+those cores, and we want to avoid mixing management and normal workloads there.
+Expanding on the already build workload partitioning we should be able to supply
+the same functionality to HA and 3NC clusters.
+
+### User Stories
+
+As a cluster creator I want to isolate the management pods of Openshift in
+compact(3NC) and HA clusters to specific CPU sets so that I can isolate the
+platform workloads from the application workload due to the high performance and
+determinism required of my applications.
+
+### Goals
+
+- This enhancement describes an approach for configuring OpenShift clusters to
+  run with management workloads on a restricted set of CPUs.
+- Clusters built in this way should pass the same kubernetes and OpenShift
+  conformance and functional end-to-end tests as similar deployments that are
+  not isolating the management workloads.
+- We want to run different workload partitioning on masters and workers.
+- Customers will be supplied with the advice of 4 hyperthreaded cores for
+  masters and for workers, 2 hyperthreaded cores.
+- We want a general approach, that can be applied to all OpenShift control plane
+  and per-node components via the PerformenceProfile
+- We want to be clear with customers that this enhancement is a day 0 supported
+  feature only. We do not support turning it off after the installation is done
+  and the feature is on.
```

Comment:
Currently we don't prevent them from changing those settings after the cluster is up. The reason for the strong wording there is to be explicit with our customers on how we intend to support this enhancement and what we can guarantee in terms of behavior.

From a technical aspect, this relates to how the pods get mutated during admission. My understanding is that If a user were to toggle this then pods will need to be re-admited to have a guarantee that the correct pod `resources` are applied and so that we can also guarantee that they are assigned to the correct CPU set.

Reply:
The partition size is configurable day 2. Disabling the feature entirely though would not be.
I think we need to discuss if we want an explicit activation API

Reply:
Hey @dhellmann, reworked the approach here a bit to make it harder to turn off the feature. Please give it another read when you have some time. Essentially, disabling this feature would now be on par with the current way we deploy partitioned clusters, so the user would need to take an active step in deleting a machine configs to turn it off. There would be no exposed API to turn it off post install.